tiered-storage/file-system/src/main/java/org/apache/bookkeeper/mledger/offload/filesystem/impl/FileStoreBackedReadHandleImpl.java

```diff
@@ -93,14 +103,20 @@ public LedgerMetadata getLedgerMetadata() {
     @Override
     public CompletableFuture<Void> closeAsync() {
         CompletableFuture<Void> promise = new CompletableFuture<>();
+
+        if (!closeFuture.compareAndSet(null, promise)) {
+            return FutureUtil.apply(closeFuture.get(), promise);
+        }
```

Comment:
why isn't `closeFuture.get()` returned directly?

Reply:
I also thought about returning `closeFuture.get()` directly, but `promise` is already created, I want to use it

Reply:
> but promise is already created, I want to use it

Why? The creation is an extremely lightweight operation. That's fine to let GC take care of that. Very short living objects like this are handled extremely efficiently by Java GC. There's no need to worry about that.

Reply:
Yes, I known, just don't want to waste it. 
return `closeFuture.get()` is better, that I don't need to add a new method to `FutureUtil`